conf/modules.config

```diff
@@ -824,127 +826,159 @@ if (!params.skip_alignment && !params.skip_qc) {
     }
 
     if (!params.skip_rseqc && params.rseqc_modules) {
-        process {
-            withName: '.*:RSEQC:RSEQC_BAMSTAT' {
-                publishDir = [
-                    path: { "${params.outdir}/${params.aligner}/rseqc/bam_stat" },
-                    mode: params.publish_dir_mode,
-                    saveAs: { filename -> filename.equals('versions.yml') ? null : filename }
-                ]
-            }
-
-            withName: '.*:RSEQC:RSEQC_INFEREXPERIMENT' {
-                publishDir = [
-                    path: { "${params.outdir}/${params.aligner}/rseqc/infer_experiment" },
-                    mode: params.publish_dir_mode,
-                    saveAs: { filename -> filename.equals('versions.yml') ? null : filename }
-                ]
-            }
-
-            withName: '.*:RSEQC:RSEQC_JUNCTIONANNOTATION' {
-                publishDir = [
-                    [
-                        path: { "${params.outdir}/${params.aligner}/rseqc/junction_annotation/pdf" },
-                        mode: params.publish_dir_mode,
-                        pattern: '*.pdf'
-                    ],
-                    [
-                        path: { "${params.outdir}/${params.aligner}/rseqc/junction_annotation/bed" },
-                        mode: params.publish_dir_mode,
-                        pattern: '*.bed'
-                    ],
-                    [
-                        path: { "${params.outdir}/${params.aligner}/rseqc/junction_annotation/xls" },
-                        mode: params.publish_dir_mode,
-                        pattern: '*.xls'
-                    ],
-                    [
-                        path: { "${params.outdir}/${params.aligner}/rseqc/junction_annotation/log" },
-                        mode: params.publish_dir_mode,
-                        pattern: '*.log'
-                    ],
-                    [
-                        path: { "${params.outdir}/${params.aligner}/rseqc/junction_annotation/rscript" },
+        if ('bam_stat' in rseqc_modules) {
```

Comment:
```suggestion
        if (!params.skip_rseqc && 'bam_stat' in rseqc_modules) {
```
Perhaps combine this with above? I prefer anything we can do to get rid of these nested if's.
Also I'm realizing it's harder to review, because it's easy to mistake which level of nesting you're in.

Reply:
Changed in https://github.com/nf-core/rnaseq/pull/771/commits/3016c0bbe6256e198eea79d9dabe3aaf4648fe09

Can you double-check it looks good for me please?

We really need to find a way to avoid this logic duplication....I will try and speak to our very own Yoda.

Reply:
The logic duplication can be avoided by a pipeline rewrite. The alternative is to use channel branching and ternary operators to select the correct input channels, along channel mixing and the `when`. It's not a simple change though. 

But it would be nicer to still be able to put process invocations in `if`'s. Maybe that's not what the data flow model was intended for though. 🤷🏽  Most people writing these workflows come from training using R or Python or what have you, and are not used to this data flow model.